src/pudl_archiver/depositors/zenodo.py

```diff
@@ -191,18 +199,21 @@ async def get_new_version(self, deposition: Deposition) -> Deposition:
             A new Deposition that is a snapshot of the old one you passed in,
             with a new major version number.
         """
+        # If we try to get a new version of a draft, delete the draft and request a fresh snapshot
         if not deposition.submitted:
             return deposition
```

Comment:
Should this call `delete_deposition` instead of returning the old draft?

Reply:
Is the idea here that even though we expect to either delete drafts (if no changes) or publish them (if changes), and thus never expect to see an unpublished draft, if something has gone wrong such that there is a draft hanging around when the archiver runs, it'll delete it and start over with a fresh snapshot of the most recently published deposition?

Reply:
If we initialize a new deposition, the first version that comes back is unsubmitted. I initially had "delete draft on sight, then ask for a new one" behavior, but we couldn't ask for a new one from a deposition with *no* versions.

Reply:
This comment is left over from when we had that behavior - sorry for the confusion!

Reply:
If we're not deleting draft depositions at the start, there's still a chance that we encounter a situation where we start working on a new version, but when we try to publish it something goes wrong and it doesn't publish so we're left with a floating draft that could cause unexpected behavior (like assuming nothing has changed). Could we delete drafts if `deposition.submitted` is False and we're running with `--initialize`?